src/tterm_printer.ml

```diff
@@ -18,15 +18,15 @@ let print_vs fmt { vs_name; vs_ty } =
   pp fmt "@[%a:%a@]" Ident.pp_simpl vs_name print_ty vs_ty
 
 let print_ls_decl fmt { ls_name; ls_args; ls_value; _ } =
-  let is_func = Option.is_some ls_value in
+  let is_func = ty_equal ls_value ty_bool in
```

Comment:
Sorry for the late comment: isn't that the other way around?

Reply:
Yes, my bad :( I've updated the code with the fixed version. Just a note: this gospel code:
`(*@ type t = {x : int -> int} *)`

Produces the following result in the pretty printer

``` 
(*@ type t = {x:int -> int}
      function constr#t (_:int -> int)t
      function x (_:t)int -> int *)```

Is this what we want? I don't think we should change this in this PR, but this behaviour seems very unusual.

Reply:
I believe we should always print the `:` as predicates know also have a type.

The pretty-printer is not really pretty, it is not even a legal printer. I believe it was first written for debugging purpose.
I'm in the process of adding a proper pretty-printer: see #407 and https://github.com/n-osborne/gospel/tree/term-pretty-printer.

Reply:
ok, the git history and the `:` are fixed

Reply:
Thanks!